.github/workflows/nomad-common.yaml

```diff
@@ -41,51 +56,45 @@ jobs:
           - write_read
           - write_validated
           - zome_call_single_value
-        include:
-          - job-name: app_install_large
-            scenario-name: app_install
-
-          - job-name: app_install_minimal
-            scenario-name: app_install
-
-          - scenario-name: dht_sync_lag
-            required-nodes: 10
-
-          - scenario-name: write_get_agent_activity
-            required-nodes: 2
-
-          - scenario-name: write_validated_must_get_agent_activity
-            required-nodes: 2
-
-          - scenario-name: zero_arc_create_data
-            required-nodes: 10
-
-          - scenario-name: zero_arc_create_data_validated
-            required-nodes: 10
-
-          - scenario-name: zero_arc_create_and_read
-            required-nodes: 9
-
-          - scenario-name: full_arc_create_validated_zero_arc_read
-            required-nodes: 10
-
-          - scenario-name: mixed_arc_get_agent_activity
-            required-nodes: 10
+          - app_install_large
+          - app_install_minimal
+          - dht_sync_lag
+          - write_get_agent_activity
+          - write_validated_must_get_agent_activity
+          - two_party_countersigning
+          - zero_arc_create_data
+          - zero_arc_create_data_validated
+          - zero_arc_create_and_read
+          - full_arc_create_validated_zero_arc_read
+          - mixed_arc_get_agent_activity
 
     env:
-      RUN_ID: "${{ matrix.job-name || matrix.scenario-name }}_${{ github.run_id }}"
-      JOB_NAME: "${{ matrix.job-name || matrix.scenario-name }}"
+      RUN_ID: ${{ matrix.job-name }}_${{ github.run_id }}
+
     steps:
       - uses: actions/checkout@v6
         with:
-          ref: "${{ inputs.commit_hash || github.sha }}"
+          ref: ${{ inputs.commit-hash || github.sha }}
+
+      - name: Read Nomad Job Variables
+        id: read-nomad-vars
+        run: |
+          JOB_VARS_FILE="${{ inputs.nomad-job-variant-directory }}/vars/${{ matrix.job-name }}.json"
+          NOMAD_VAR_DURATION="$(jq -e -r '.duration' "$JOB_VARS_FILE")"
+          NOMAD_VAR_SCENARIO_NAME="$(jq -e -r '.scenario_name' "$JOB_VARS_FILE")"
+          # Parse required nodes count from length of 'behaviours' list, or else default to 1.
+          REQUIRED_NODES="$(jq -e -r '.behaviours // 1 | length' "$JOB_VARS_FILE")"
```

Comment:
The minimum number of nodes is the length of the "behaviours" array in the nomad vars file, or else defaults to 1.